test/EthernetServerTest.cpp

```diff
@@ -44,8 +46,123 @@ unittest(basic) {
+  server->loop();
+  deque<uint8_t>* pBuffer = client.writeBuffer();
+  assertTrue(pBuffer->size() > 100);
+  String response;
+  while (!pBuffer->empty()) {
+    response.concat(pBuffer->front());
+    pBuffer->pop_front();
+  }
+  const char expectedResponse[] =
+      "HTTP/1.1 200 OK\r\n"
+      "Content-Type: text/plain;charset=UTF-8\r\n"
+      "Content-Encoding: identity\r\n"
+      "Content-Language: en-US\r\n"
+      "Content-Length: 36\r\n"
+      "\r\n"
+      "pH=0.000   8.100\r\n"
+      "T= 0.00 h 20.00 \r\n";
+  assertEqual(expectedResponse, response);
+  assertEqual(NOT_CONNECTED, server->getState());
+  client.stop();
+  server->loop();
+}
+
+unittest(keypress) {
+  TankController* tc = TankController::instance();
+  LiquidCrystal_TC* lcd = LiquidCrystal_TC::instance();
+  assertEqual("MainMenu", tc->stateName());
+  EthernetServer_TC* server = EthernetServer_TC::instance();
+  EthernetClient_CI client;
+  server->setHasClientCalling(true);
+  delay(1);
+  server->loop();
+  client = server->getClient();
+  const char request[] =
+      "POST /api/1/key?value=2 HTTP/1.1\r\n"
+      "Host: localhost:80\r\n"
+      "Accept: text/plain;charset=UTF-8\r\n"
+      "Accept-Encoding: identity\r\n"
+      "Accept-Language: en-US\r\n"
+      "\r\n";
+  client.pushToReadBuffer(request);
+  server->loop();
+  deque<uint8_t>* pBuffer = client.writeBuffer();
+  assertEqual(52, pBuffer->size());
+  String response;
+  while (!pBuffer->empty()) {
+    response.concat(pBuffer->front());
+    pBuffer->pop_front();
+  }
+  const char expectedResponse[] =
+      "HTTP/1.1 303 See Other\r\n"
+      "Location: /api/1/display\r\n"
+      "\r\n";
+  assertEqual(expectedResponse, response);
+  tc->loop();  // Loop to handle the UI press
+  assertEqual("Change settings ", lcd->getLines().at(0));
+  delay(60000);  // IDLE_TIMEOUT
+  tc->loop();
+  assertEqual("MainMenu", tc->stateName());
+  assertEqual(NOT_CONNECTED, server->getState());
+  client.stop();
+  server->loop();
+}
+
+unittest(badRequest) {
+  TankController* tc = TankController::instance();
+  EthernetServer_TC* server = EthernetServer_TC::instance();
+  EthernetClient_CI client;
+  server->setHasClientCalling(true);
+  delay(1);
+  server->loop();
+  client = server->getClient();
+  const char request[] =
+      "POST /api/1/key?value=foo HTTP/1.1\r\n"
+      "Host: localhost:80\r\n"
+      "Accept: text/plain;charset=UTF-8\r\n"
+      "Accept-Encoding: identity\r\n"
+      "Accept-Language: en-US\r\n"
+      "\r\n";
+  client.pushToReadBuffer(request);
+  server->loop();
+  deque<uint8_t>* pBuffer = client.writeBuffer();
+  assertTrue(pBuffer->size() == 28);
+  String response;
+  while (!pBuffer->empty()) {
+    response.concat(pBuffer->front());
+    pBuffer->pop_front();
+  }
+  const char expectedResponse[] = "HTTP/1.1 400 Bad Request\r\n\r\n";
+  assertEqual(expectedResponse, response);
+  assertEqual(NOT_CONNECTED, server->getState());
+  client.stop();
+  server->loop();
+}
+
```

Comment:
Add test for current endpoint.

Reply:
Were you referring to the unit test for display, or did you have something else in mind?

Reply:
Test for current values added to EthernetServerTest.cpp